File: 0495-teemo-attacking/0495-teemo-attacking.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {

        int poisoned = 0;
        int nextWakeup = 0;

        for(int time : timeSeries) {
            if (time < nextWakeup) poisoned -= nextWakeup - time;
            poisoned += duration;
            nextWakeup = time + duration;
        }

        return poisoned;
    }
}

File: 0495-teemo-attacking/0495-teemo-attacking.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        if not timeSeries:
            return 0
        
        total = 0
        start = timeSeries[0]
        end = start + duration
        
        for i in range(1, len(timeSeries)):
            time = timeSeries[i]
            if time > end:
                total += end-start  # add to total and reset start
                start = time
            end = time + duration
        
        total += end - start  # must account for final time in timeSeries
        return total